library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -71,8 +71,42 @@ internal class UploadMetrics private constructor() {
 
   private suspend fun sendPost(body:JSONObject) {
     val httpClient = MuxUploadSdk.httpClient()
+      .newBuilder()
+      // The SDK's http client is configured for uploading. We want the tighter default timeouts
+      .callTimeout(0, TimeUnit.SECONDS)
+      .writeTimeout(10, TimeUnit.SECONDS)
+      // We need to do a non-compliant redirect: 301 or 302 but preserving method and body
+      .addInterceptor(Interceptor { chain ->
+        val response = chain.proceed(chain.request())
+        if (response.code in 301..302 || response.code in 307..308) {
+          val redirectUri = response.headers("Location").firstOrNull()?.let { Uri.parse(it) }
+          // If 'Location' was present and was a real URL, redirect to it
+          if (redirectUri == null) {
+            logger.w("UploadMetrics", "redirect with invalid or blank url. ignoring")
+            response
+          } else {
+            response.close() // Required before starting a new request in the chain
+            val redirectedReq = response.request.newBuilder()
+            if (redirectUri.isRelative) {
+              val requestUrl = response.request.url
+              var relativePathDelimiter = "/"
+              if (redirectUri.toString().startsWith("/")) {
+                relativePathDelimiter = "";
+              }
+              redirectedReq.url(requestUrl.scheme + "//" + requestUrl.host + ":"
+                      + requestUrl.port + relativePathDelimiter + redirectUri.toString())
+            } else {
+              redirectedReq.url(redirectUri.toString())
+            }
+            chain.proceed(redirectedReq.build())
+          }
+        } else {
+          response
+        }
+      })
+      .build()
     val request = Request.Builder()
-      .url("https://mobile.muxanalytics.com")
+      .url("https://mux-sdks-telemetry.vercel.app/api/upload-sdk-native")
```

Comment:
nitpick: We have already updated the CNAME mapping, so this should be

```suggestion
      .url("https://mobile.muxanalytics.com/api/upload-sdk-native")
```